### ISSUE TITLE
ReplacePathRegex could possibly set invalid values for X-Replaced-Path header and cause internal server error

### ISSUE THREAD

**Issue by happydenn** [traefik/traefik#8228]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

We have a Traefik instance that's acting as a reverse proxy to a backend API server:

```
     /api/abc            replacePath -> /abc
req ----------> Traefik ---------------------> API
```

We expect that all requests send to `http://traefik/api/*` would be proxied to `http://api/*`, which is working for the majority of requests.

### What did you see instead?

However we started to observe a percentage of 500 errors showing up on our monitoring system, which we finally drilled down to a problem with the `ReplacePathRegex` middleware.

Here's an example request path that caused the error: `/api/%08Claire%20Wu`

It was puzzling us at first because we had no problem access the API server directly with this path: `/%08Claire%20Wu`

Further investiation showed that none of our requests that resulted in 500 had reach our API, then we turned on Traefik's debug log and then the root cause of this issue is clear in sight:

```
DEBU[2021-06-24T11:30:12Z] '500 Internal Server Error' caused by: net/http: invalid header field value "/api/\bClaire Wu" for key X-Replaced-Path
```

It seemed like the value that `ReplacePathRegex` set contains a backspace character (\b) which is invalid for header value and the underlying Go http library was throwing the error (ref. [net/http code throwing the error](https://github.com/golang/go/blob/5c489514bc5e61ad9b5b07bd7d8ec65d66a0512a/src/net/http/transport.go#L527)). This choked Traefik's proxy pipeline and resulted in 500 internal server error.

As the path is URL encoded and completely legal, this worried us that maybe there are other requests failing because of this as path is provided by the client and Traefik couldn't handle a valid request where the upstream server could.

Currently we worked around this issue using an additional `Headers` middleware to strip the `X-Replaced-Path` header set by `ReplacePathRegex`.

### What version of Traefik are you using?

v2.4.8

### What is your environment & configuration?

```yaml
http:
  routers:
    api:
      service: api
      rule: PathPrefix(`/api/`)
      middlewares:
      - replace-path

  services:
    api:
      loadBalancer:
        servers:
        - url: http://api

  middlewares:
    replace-path:
      replacePathRegex:
        regex: "^/api/(.*)"
        replacement: "/$1"
```

Traefik is running as the ingress controller for a Kubernetes cluster. I've extracted the important bits above.

Below is the middleware that worked around this issue:
```yaml
headers:
  customRequestHeaders:
    X-Replaced-Path: ""
```

**Comment by dtomcej** [comment]
It looks like the offending code is here:

https://github.com/traefik/traefik/blob/master/pkg/middlewares/replacepathregex/replace_path_regex.go#L61

Where we set the `X-Replaced-Path` header based on the pre-modified path. It appears that our check to see if the path was encoded is not triggering properly. Perhaps an additional check for validity could be added here, and in the corresponding code in the ReplacePath middleware.

**Comment by dtomcej** [comment]
Also of note, according to w3schools: (https://www.w3schools.com/tags/ref_urlencode.ASP)

`%08` is the URL encoding for `backspace`.

**Comment by tomMoulard** [comment]
Closed by #8251
